testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/ClientClusterNodeSelectorTestCase.java

```diff
@@ -0,0 +1,219 @@
+ * client (refer to issue <a href="https://issues.jboss.org/browse/WFLY-10030">WFLY-10030</a>); that is why a Byteman rule
+ * has been added to track the affinity and print it out in case the test fails;</p>
+ * <p>NOTE: this test can be run using the following command issued from the root directory of the whole WildFly project:
+ * {@code ./integration-tests.sh test -Dts.noSmoke -Dts.clustering -Dtest=org.jboss.as.test.clustering.cluster.ejb.remote.ClientClusterNodeSelectorTestCase}</p>
+ *
+ * @author <a href="mailto:tborgato@redhat.com">Tommaso Borgato</a>
+ */
+@RunWith(Arquillian.class)
+@BMUnitConfig(loadDirectory="target/test-classes")
+@BMScript(value= "clientclusternodeselector.btm")
+public class ClientClusterNodeSelectorTestCase extends AbstractClusteringTestCase {
+
+    private static final String MODULE_NAME = "remote-stateless-ejb-cns-test";
+    private static final long CLIENT_TOPOLOGY_UPDATE_WAIT = TimeoutUtil.adjust(2000);
+    private static final String PROPERTIES_FILE = "clustered-jboss-ejb-client.properties";
+
+    public static String AFFINITY;
+    public static String CLUSTER_NAME;
+
+    /**
+     * Implementation of {@link ClusterNodeSelector} to be used for custom cluster node selection
+     */
+    public static class CustomClusterNodeSelector implements ClusterNodeSelector {
+        private static volatile String PICK_NODE = null;
+
+        @Override
+        public String selectNode(
+                String clusterName, String[] connectedNodes, String[] totalAvailableNodes) {
+            if (PICK_NODE != null) {
+                return PICK_NODE;
+            }
+            return connectedNodes[0];
+        }
+    }
+
+    public ClientClusterNodeSelectorTestCase() throws Exception {
+        super(FOUR_NODES);
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        // calls into the byteman engine to do the setup from annotations on this class
+        BMUnitConfigState.pushConfigurationState(
+                ClientClusterNodeSelectorTestCase.class.getAnnotation(BMUnitConfig.class)
+                , ClientClusterNodeSelectorTestCase.class);
+        BMUnit.loadScriptFile(
+                ClientClusterNodeSelectorTestCase.class
+                , BMRunnerUtil.computeBMScriptName(ClientClusterNodeSelectorTestCase.class.getAnnotation(BMScript.class).value())
+                , BMRunnerUtil.normaliseLoadDirectory(ClientClusterNodeSelectorTestCase.class.getAnnotation(BMScript.class)));
+
+        // trigger the static init of the correct properties file - this also depends on running in forkMode=always
+        JBossEJBProperties ejbProperties =
+                JBossEJBProperties.fromClassPath(ClientClusterNodeSelectorTestCase.class.getClassLoader()
+                        , PROPERTIES_FILE);
+        JBossEJBProperties.getContextManager().setGlobalDefault(ejbProperties);
```

Comment:
Should this be cleaned up after the test has run?

Reply:
Those statements can also be removed completely, I just realized that the clustered-jboss-ejb-client.properties isn't used at all